src/FileStorageBase/Invoker/ContractInvoker.cs

```diff
@@ -0,0 +1,133 @@
+                ValidUntilBlock = height + NeoSystem.Settings.MaxValidUntilBlockIncrement,
+                Signers = new Signer[] { new Signer() { Account = Wallet.GetAccounts().ToArray()[0].ScriptHash, Scopes = WitnessScope.Global } },
+                Attributes = Array.Empty<TransactionAttribute>(),
+            };
+            tx.SystemFee = result.GasConsumed + fee;
+            tx.NetworkFee = Wallet.CalculateNetworkFee(snapshot, tx);
+            var balance = NativeContract.GAS.BalanceOf(snapshot, Wallet.GetAccounts().First().ScriptHash);
+            if (tx.SystemFee + tx.NetworkFee > balance)
+            {
+                Utility.Log(nameof(ContractInvoker), LogLevel.Debug, $"insufficient gas, network={NeoSystem.Settings.Network}, method={method}, height={height}, system_fee={tx.SystemFee}, network_fee={tx.NetworkFee}, balance={balance}");
+                throw new InvalidOperationException($"make transaction failed, gas insufficient");
+            }
+            var data = new ContractParametersContext(snapshot, tx, NeoSystem.Settings.Network);
+            if (!Wallet.Sign(data)) throw new InvalidOperationException($"make transaction failed, wallet could not sign");
+            tx.Witnesses = data.GetWitnesses();
+            Blockchain.Tell(tx);
+            Utility.Log(nameof(ContractInvoker), LogLevel.Debug, $"tx sent, network={NeoSystem.Settings.Network}, tid={tx.Hash}, height={height}, method={method}");
+        }
+
+        protected InvokeResult TestInvoke(UInt160 contractHash, string method, params object[] args)
+        {
+            byte[] script = contractHash.MakeScript(method, args);
+            FakeSigners signers = new(Wallet.GetAccounts().First().ScriptHash);
+            var result = GetInvokeResult(script, signers);
+            if (result.State != VMState.HALT)
+            {
+                Utility.Log(nameof(ContractInvoker), LogLevel.Error, $"perform invoke failed, method={method}, error={result.FaultException.Message}");
+                throw new InvalidOperationException($"perform invoke failed, method={method}");
+            }
+            return result;
+        }
+
+        protected InvokeResult GetInvokeResult(byte[] script, FakeSigners signers = null)
+        {
+            using SnapshotCache snapshot = NeoSystem.GetSnapshot();
+            ApplicationEngine engine = ApplicationEngine.Run(script, snapshot, container: signers, null, NeoSystem.Settings, 0, 20000000000);
+            return new InvokeResult()
+            {
+                State = engine.State,
+                GasConsumed = engine.GasConsumed,
+                Script = script,
+                FaultException = engine.FaultException,
+                UncaughtException = engine.UncaughtException,
+                ResultStack = engine.ResultStack.ToArray()
+            };
+        }
+
+        public void TransferGas(UInt160 to, long amount)
+        {
+            var account = Wallet.GetAccounts().ToArray()[0].ScriptHash;
```

Comment:
We don't have a reorder in wallet but we have `IsDefault`, we should find the default one instead of check always the first one

Reply:
Update. Use default or first.